pyo3-build-config/src/impl_.rs

```diff
@@ -321,6 +325,12 @@ print("ext_suffix", get_config_var("EXT_SUFFIX"))
             .parse()
             .context("failed to parse calcsize_pointer")?;
 
+        let build_flags: BuildFlags = BuildFlags::from_interpreter(interpreter)?;
+
+        if build_flags.0.contains(&BuildFlag::Py_GIL_DISABLED) && abi3 {
+            bail!("Cannot set Py_LIMITED_API and Py_GIL_DISABLED at the same time")
+        }
```

Comment:
I think in some cases we simply warn and disable `Py_LIMITED_API` (e.g., PyPy), do we have a principled reason to do one vs. the other?

Reply:
In 3.13 `Py_GIL_DISABLED` implies `Py_LIMITED_API` isn't set I wanted a way to enforce that. Warning and disabling the limited API gives the same effect.

I see there's early-exits in some of these functions for pypy and graal. I think can do something similar for free-threaded.